presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java

```diff
@@ -853,7 +857,18 @@ private void assertRead(Optional<HivePageSourceFactory> pageSourceFactory, Optio
             assertNotNull(session, "session must be specified");
             assertTrue(rowsCount >= 0, "rowsCount must be greater than zero");
 
-            File file = File.createTempFile("presto_test", formatName);
+            String compressionSuffix = compressionCodec.getCodec()
+                    .map(codec -> {
+                        try {
+                            return codec.newInstance().getDefaultExtension();
```

Comment:
Avoid `Class.newInstance()` since it can throw undeclared checked exceptions
```java
return codec.getConstructor().newInstance().getDefaultExtension();